recipes/iceoryx/all/conanfile.py

```diff
@@ -0,0 +1,244 @@
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["TOML_CONFIG"] = self.options.toml_config
+        self._cmake.configure()
+        return self._cmake
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True,
+                  destination=self._source_subfolder)
+
+    def build(self):
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy("LICENSE", src=self._source_subfolder, dst="licenses")
+        tools.rmdir(self._pkg_share)
+        tools.rmdir(self._pkg_cmake)
+        tools.mkdir(self._pkg_res)
+        if self.options.toml_config:
+            tools.rename(
+                os.path.join(self._pkg_etc, "roudi_config_example.toml"),
+                os.path.join(self._pkg_res, "roudi_config.toml")
+            )
+        tools.rmdir(self._pkg_etc)
+        for alias, aliased in self._target_aliases.items():
+            cmake_file = "conan-official-{}-targets.cmake".format(
+                aliased.replace("::", "_")
+            )
+            self._create_cmake_module_alias_targets(
+                os.path.join(
+                    self.package_folder,
+                    self._module_subfolder,
+                    cmake_file
+                ),
+                alias,
+                aliased
+            )
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "iceoryx"
+        self.cpp_info.names["cmake_find_multi_package"] = "iceoryx"
+        # platform component
+        self.cpp_info.components["platform"].name = "platform"
+        self.cpp_info.components["platform"].libs = ["iceoryx_platform"]
+        if self.settings.os in ["Linux","Macos","Neutrino"]:
+            self.cpp_info.components["platform"].system_libs.append("pthread")
+        # utils component
+        self.cpp_info.components["utils"].name = "utils"
+        self.cpp_info.components["utils"].libs = ["iceoryx_utils"]
+        self.cpp_info.components["utils"].requires = ["platform"]
+        if self.settings.os == "Linux":
+            self.cpp_info.components["utils"].requires.append("acl::acl")
+            self.cpp_info.components["utils"].system_libs.append("rt")
+        if self.settings.os in ["Linux","Macos","Neutrino"]:
+            self.cpp_info.components["utils"].system_libs.append("pthread")
+        if self.settings.os == "Linux":
+            self.cpp_info.components["utils"].system_libs.append("atomic")
+        self.cpp_info.components["utils"].builddirs = self._pkg_cmake
+        self.cpp_info.components["utils"].build_modules["cmake_find_package"] = [
+            os.path.join(self._module_subfolder, "conan-official-iceoryx_utils-targets.cmake")
+        ]
```

Comment:
Could we not follow the usual template with one file per project?

It's a lot less code bloat

https://github.com/conan-io/conan-center-index/blob/41d4fbb8c218f9d938ccb8b9d888bff52002e62e/recipes/openmesh/all/conanfile.py#L113